### PR TITLE
Fix Permission Management with One-Way Peering

### PR DESCRIPTION
**Pull request by aleoli**
# Description

This pr fixes an issue, related to missing permission, that prevented the successful establishment of a one-way peering.

Now the ForeignCluster Operator handles the permission to assign to the remote cluster and enforces their status.

# How Has This Been Tested?

- [x] locally on KinD
- [x] add unit test

### CODE REVIEW

File: internal/discovery/foreign-cluster-operator/foreign-cluster-operator_test.go [hunk 1]
```diff
@@ -642,4 +644,240 @@ var _ = Describe("ForeignClusterOperator", func() {
...
+						},
+						Namespace:     "liqo",
+						DiscoveryType: discovery.ManualDiscovery,
+						AuthURL:       "",
+						TrustMode:     discovery.TrustModeUntrusted,
+					},
+					Status: discoveryv1alpha1.ForeignClusterStatus{
+						TenantControlNamespace: discoveryv1alpha1.TenantControlNamespace{},
+					},
+				},
```
Comment (palexster) [inline internal/discovery/foreign-cluster-operator/foreign-cluster-operator_test.go:758]: Those values are repeated and can be put as constants.

File: internal/discovery/foreign-cluster-operator/foreign-cluster-operator_test.go [hunk 2]
```diff
@@ -642,4 +636,245 @@ var _ = Describe("ForeignClusterOperator", func() {
...
+					Name: "liqo-outgoing",
+				},
+			}
+			clusterRole2 = rbacv1.ClusterRole{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "liqo-incoming",
+				},
+			}
+
+			Expect(controller.Client.Create(ctx, &clusterRole1)).ToNot(HaveOccurred())
```
Comment (giorio94) [inline internal/discovery/foreign-cluster-operator/foreign-cluster-operator_test.go:669]:
```suggestion
			Expect(controller.Client.Create(ctx, &clusterRole1)).To(Succeed())
```

Nit. Just a bit more self explanatory (and shorter)

File: internal/discovery/foreign-cluster-operator/foreign-cluster-operator_test.go [hunk 3]
```diff
@@ -642,4 +636,245 @@ var _ = Describe("ForeignClusterOperator", func() {
...
+			}
+		})
+
+		JustAfterEach(func() {
+			var roleBindingList rbacv1.RoleBindingList
+			Expect(controller.Client.List(ctx, &roleBindingList)).ToNot(HaveOccurred())
+			for i := range roleBindingList.Items {
+				rb := &roleBindingList.Items[i]
+				Expect(controller.Client.Delete(ctx, rb)).ToNot(HaveOccurred())
+			}
```
Comment (giorio94) [inline internal/discovery/foreign-cluster-operator/foreign-cluster-operator_test.go:689]: Nit. Could be probably replaced by a `controller.Client.DeleteAllOf(...)` call

**Review by giorio94 — APPROVED** [review]
/lgtm

**Review by palexster — APPROVED** [review]

File: internal/discovery/foreign-cluster-operator/foreign-cluster-operator_test.go (hunk 3 above)
Comment (aleoli) [reply internal/discovery/foreign-cluster-operator/foreign-cluster-operator_test.go:689]: No, it tells me that the `deletecollection` verb is not allowed by this API Server on that resource

Comment (giorio94) [reply internal/discovery/foreign-cluster-operator/foreign-cluster-operator_test.go:689]: Ah ok, didn't know about that

**Pull request merged**